src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -961,12 +961,38 @@ public override BoundNode VisitPassByCopy(BoundPassByCopy node)
 
         public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node)
         {
-            Debug.Assert(!IsConditionalState);
-            VisitRvalue(node.Expression);
-
             bool negated = node.Pattern.IsNegated(out var pattern);
             Debug.Assert(negated == node.IsNegated);
 
+            Debug.Assert(!IsConditionalState);
```

Comment:
Small nit: this could still be at the start of the function.

Reply:
Done